Sources/HexColorTransform.swift

```diff
@@ -54,10 +54,15 @@ open class HexColorTransform: TransformType {
 	
 	fileprivate func hexString(color: Object) -> String {
 		let comps = color.cgColor.components!
+		let compsCount = color.cgColor.numberOfComponents
```

Comment:
My 2¢ here would be to `switch` on the number of components instead. It will better convey the purpose of the branching logic and how it is handled:

```
let r: Int
let g: Int
let b: Int
let a: Int
switch compsCount {
case 2: // Grayscale
    r = ..
    ...
case 4: // rgba
   r = ...
}
```